HealthcareSystem/PatientWebApp/Controllers/SurveyController.cs

```diff
@@ -20,29 +21,39 @@ public class SurveyController : ControllerBase
         private readonly ISurveyService _surveyService;
         private readonly SurveyValidator _surveyValidator;
 
-        public SurveyController(ISurveyService surveyService)
+        private readonly IExaminationService _examinationService;
+        private readonly ExaminationValidator _examinationValidator;
+
+
+        public SurveyController(ISurveyService surveyService, IExaminationService examinationService)
         {
             _surveyService = surveyService;
             _surveyValidator = new SurveyValidator(surveyService);
+            _examinationService = examinationService;
+            _examinationValidator = new ExaminationValidator(_examinationService);
         }
 
+
         [HttpPost]
         public ActionResult AddSurvey(SurveyDTO surveyDTO)
         {
             try
             {
                 _surveyValidator.ValidateSurveyFields(surveyDTO);
+                _examinationValidator.CheckIfExaminationIsFinished(surveyDTO.ExaminationId);
+                _examinationValidator.CheckIfSurveyAboutExaminationIsCompleted(surveyDTO.ExaminationId);
                 _surveyService.AddSurvey(SurveyMapper.SurveyDTOToSurvey(surveyDTO));
+                _examinationService.CompleteSurveyAboutExamination(surveyDTO.ExaminationId);
+                return Ok();
```

Comment:
I'm not a fan of the fact that basically, the Controller is doing orchestration here, not the Services. I'm also not a fan of the fact that you're fetching the Examination from the Repository three separate times, twice by the Validator and once by the Service. Basically, here's approximately what I think should be happening - this is all done by the SurveyService instead. And the method in SurveyService would do something like this - fetch the Examination, call a method on the Examination to mark that a survey has been completed for it (this method in Examination would check if the examination is finished and that a survey isn't already completed before setting the field), instantiate the Survey objects from the DTO (possibly with some validation happening in the constructor) and add them to the database, update the Examination in the database.
However, my idea of what should be happening is fairly tentative right now, I would probably lave this as is until we go over DDD and then refactor.

One thing that I would like to be fixed right now on the other hand - you're not handling what happens if no Examination with this id exists. CheckIfExaminationIsFinished, CheckIfSurveyAboutExaminationIsCompleted, and CompleteSurveyAboutExamination will all end up throwing NullPointerExceptions if the Repository returns null, and it's not replaced by a local exception or handled anywhere.